### DISCUSSION TITLE
How to use Alt+backspace for delete word backward?(as most of terminal in linux)

### DISCUSSION THREAD

**Discussion by zw963** [kovidgoyal/kitty#3476] [Q&A] [answered]
As described in this question:

https://unix.stackexchange.com/a/94332/148127

When use with kitty, i have to get same behavior with `alt + shift + backspace`.

is there a clue about how to rebind above to `alt+backsapce`?

and how to get the full action list for kitty?

Thank you.

**Comment by kovidgoyal** [comment, marked as answer]
This is a function of shells not terminals. Use showkey -a to see what bytes are produced by alt+backspace and bind them to the delete word function in your shells rc files.

And running showkey -a you will find that the terminal generates the exact same bytes for both alt+backspace and alt+shift+backspace.

**Reply by zw963** [reply]
> And running showkey -a you will find that the terminal generates the exact same bytes for both alt+backspace and alt+shift+backspace.

Yes, i find it now, previous, because some keybinding mapping application is running, which cause both not same, sorry
for confuse, in fact, `alt + backspace` just work fine.

**Comment by zw963** [comment]
BTW,  @kovidgoyal  can you please help me on how to get the full action name list which can be map to a hotkey?

Thank you.

**Comment by glensc** [comment]
the year 2023 update: The `showkey` command I assume is `kitty +kitten show_key -a`, the `-a` option no longer works (kitty 0.27.1), but without `-a` it's useful or `-m kitty`

but for me I needed to add to `.zshrc` these lines to restore alt+backspace behavior from iterm:

```sh
    # Why are these bindings missing?
    builtin bindkey "^A" beginning-of-line
    builtin bindkey "^E" end-of-line
    builtin bindkey "^[^H" backward-kill-word
    builtin bindkey "^[^?" backward-kill-word
```

I'm actually puzzled, why `zsh` in kitty lacks basic key bindings. it should invoke the same `.zprofile`, `.zshrc` files...

**Reply by page-down** [reply]
@glensc
> The showkey command ... the -a option no longer works ...

No, this refers to the `showkey` executable from the package `kbd`.

https://www.man7.org/linux/man-pages/man1/showkey.1.html
https://github.com/legionus/kbd/blob/v2.5.1/src/showkey.c

Using show_key kitten is recommended.
```shell
kitty +kitten show_key
kitty +kitten show_key -m kitty
```

> ... why zsh in kitty lacks basic key bindings ...
> ... it should invoke the same ... .zshrc files...

Mostly in which shellrc file they are unbound or changed, you can start with empty configuration files, since these are all default keys, even without loading any shellrc.

The keys will work properly when the TERM is configured correctly (`xterm-kitty`, no changes to the default) and terminfo is installed correctly.
